tests/openQA_containers/multiple_container_webui.pm

```diff
@@ -0,0 +1,74 @@
+ use strict;
+ use base "openQAcoretest";
+ use testapi;
+ 
+ sub wait_for {
+     my ($container) = @_;
+     my $count = 30;
+     while (system("docker logs $container 2>&1 | grep Listening >/dev/null") != 0) {
+         sleep 1;
+         $count = $count - 1;
+         last if ($count == 0);
+     }
+ 
+     system("docker logs $container 2>&1 | grep Listening >/dev/null");
+ }
+ 
+ sub run {
+     my ($self) = @_;
+     my $volumes = "-v \"/root/data/factory:/data/factory\" -v \"/root/data/tests:/data/tests\" -v \"/root/openQA/container/webui/conf:/data/conf:ro\"";
+ 
+     assert_script_run("mkdir -p /root/data/factory/{iso,hdd,other} /root/data/tests");
+     assert_script_run("docker network create testing");
+     assert_script_run("docker run -d --network testing -e POSTGRES_PASSWORD=openqa -e POSTGRES_USER=openqa -e POSTGRES_DB=openqa --net-alias=db --name db postgres", timeout => 600);
+ 
+     my $count = 30;
+     while (system("docker logs db 2>&1 | grep \"database system is ready to accept connections\" >/dev/null") != 0) {
+         sleep 1;
+         $count = $count - 1;
+         last if ($count == 0);
+     }
+ 
+     assert_script_run("docker logs db 2>&1 | grep \"database system is ready to accept connections\" >/dev/null");
+ 
+     #RUN webui
+     assert_script_run("docker run -d --network testing -e MODE=webui -e MOJO_LISTEN=http://0.0.0.0:9526 $volumes -p 9526:9526 --name webui openqa_webui");
+     wait_for("webui");
+     assert_script_run("docker logs webui");
+     assert_script_run("docker exec webui curl localhost:9526 >/dev/null");
+     record_info("webui working");
+     
+     #RUN websockets
+     assert_script_run("docker run -d --network testing -e MODE=websockets -e MOJO_LISTEN=http://0.0.0.0:9527 $volumes -p 9527:9527 --name websockets openqa_webui");
+     wait_for("websockets");
+     assert_script_run("docker exec websockets curl localhost:9527 >/dev/null");
+     record_info("websockets working");
+ 
+     #RUN livehandler
+     assert_script_run("docker run -d --network testing -e MODE=livehandler -e MOJO_LISTEN=http://0.0.0.0:9528 $volumes -p 9528:9528 --name livehandler openqa_webui");
+     wait_for("livehandler");
+     assert_script_run("docker exec livehandler curl localhost:9528 >/dev/null");
+     record_info("livehandler working");
+ 
+     #RUN scherduler
+     assert_script_run("docker run -d --network testing -e MODE=scheduler -e MOJO_LISTEN=http://0.0.0.0:9529 $volumes -p 9529:9529 --name scheduler openqa_webui");
+     wait_for("scheduler");
+     assert_script_run("docker exec scheduler curl localhost:9529 >/dev/null");
+     record_info("scheduler working");
```

Comment:
kind repetitive code from L35. What would you say to make a function for it?

Reply:
Done